lib/views/pay/pay_view.dart

```diff
@@ -27,10 +32,36 @@ class _PayViewState extends State<PayView> {
   String? createdTime;
   String? expirationTime;
 
-  Future<AppPayInvoice> payInvoice(String boltString) async {
-    final response =
-        await widget.provider.get<AppApi>().payInvoice(invoice: boltString);
-    return response;
+  Future<void> payInvoice(String boltString) async {
+    AppPayInvoice? response;
+    try {
+      response =
+          await widget.provider.get<AppApi>().payInvoice(invoice: boltString);
+    } catch (e) {
+      late ErrorDecoder decoder;
+      var jsonString = e.toString().substring(e.toString().indexOf('{'));
+      if (e is Exception || e is LNClientException) {
+        decoder = ErrorDecoder.fromJSON(jsonDecode(jsonString));
+      }
+      if (context.mounted) {
+        GlobalComponent.showAppDialog(
+            context: context,
+            title: 'Payment failed',
+            message: decoder.message,
+            closeMsg: 'Ok',
+            imageProvided: const AssetImage('assets/images/exclamation.png'));
+      }
+    }
+    if (response != null) {
+      if (context.mounted) {
```

Comment:
two question

1, Why do we always need this `contex.mounted`?
2. If really needed why do not put it at the beginning of the function if it is not mounted you return and nothing happens?

Reply:
The function call to the async function in the above part which is `await widget.provider.get<AppApi>().payInvoice(invoice: boltString);`. Thus, if we do not apply this, we get this warning.
`Don't use 'BuildContext's across async gaps.`

Reply:
Right I forget that dart it is painful!

Reply:
Each time after `async` function is called.